task/tekton-operator-install/0.1/tekton-operator-install.yaml

```diff
@@ -0,0 +1,187 @@
+            -o jsonpath='{range .status.conditions[0]}{.version}{"\n"}{end}'
+        }
+
+        show_failure() {
+          echo "failed to install $1 $2"
+          exit 1
+        }
+
+        main(){
+          kubectl apply -f https://storage.googleapis.com/tekton-releases/operator/$(params.OPERATOR_VERSION)/release.notags.yaml
+          local maxloop=30 # 5 minutes max
+
+          # check operator and pipeline correctly installed or not
+          # display the version of pipeline
+          local count=0
+          while true; do
+            # hard timeout condition
+            [[ ${count} == ${maxloop} ]] && show_failure "tektonpipelines" "cluster"
+
+            local codeStatus=$(checkCodeStatus "tektonpipelines" "cluster")
+            [[ "$codeStatus" == "installed" ]] && \
+            checkVersionStatus "tektonpipelines" "cluster" && break
+            sleep 10
+            count=$((count+1))
+          done
+          echo "successfully installed operator version $(params.OPERATOR_VERSION)"
+        }
+
+        main "$@"
+
+    - name: install-addon
+      image: lachlanevenson/k8s-kubectl@sha256:3a5e22a406a109f4f26ec06b5f1f6a66ae0cd0e185bc28499eb7b7a3bbf1fe09
+      workingDir: $(workspaces.kubeconfig.path)
+      script: |
+        [[ -f ./kubeconfig ]] && \
+        export KUBECONFIG=./kubeconfig || \
+        (echo "please provide kubeconfig file" && exit 1)
+
+        # create TektonAddon CR manifest and apply that directly
+        createCustomResource() {
+          cr_name=${1}
+          cat <<EOF | kubectl apply -f -
+        apiVersion: operator.tekton.dev/v1alpha1
+        kind: TektonAddon
+        metadata:
+          name: ${cr_name}
+        EOF
+        }
+
+        # check whether CR is installed or not
+        checkCodeStatus() {
+          local crd_name=${1}
+          local cr_name=${2}
+          kubectl get "${crd_name}" "${cr_name}" \
+            -o jsonpath='{range .status.conditions[0]}{.code}{"\n"}{end}'
+        }
+
+        # check the version of the installed CR
+        checkVersionStatus() {
+          local crd_name=${1}
+          local cr_name=${2}
+          kubectl get "${crd_name}" "${cr_name}" \
+            -o jsonpath='{range .status.conditions[0]}{.version}{"\n"}{end}'
+        }
+
+        show_failure() {
+          echo "failed to install $1 $2"
+          exit 1
+        }
+
+        main() {
+          local maxloop=30 # 5 minutes max
+          local count=0
+          local trigger_success=true
+          local dashboard_success=true
+          local extension_success=true
+
+          [[ $(params.INSTALL_TRIGGERS) == "false" ]] && \
+          [[ $(params.INSTALL_DASHBOARD) == "false" ]] && \
+          [[ $(params.INSTALL_EXTENSION_WEBHOOKS) == "false" ]] && \
+          echo "No addons to be installed" && exit 0
+
+          [[ $(params.INSTALL_TRIGGERS) == "true" ]] && \
+          createCustomResource "trigger" && trigger_success=false
```

Comment:
if the default of `trigger_success` is false we won't have to set it to `false` again here